test/test_binary_ufuncs.py

```diff
@@ -1183,6 +1183,34 @@ def test_binary_ops_with_scalars(self, device):
                         self.assertEqual(expected, python_op(first, second))
                         self.assertEqual(expected, torch_op(first, second))
 
+    # Tests that the function and its (array-accepting) reference produce the same
+    #   values on a range of tensors, including empty tensors, scalar tensors,
+    #   1D tensors and a large 2D tensor
+    @onlyOnCPUAndCUDA
+    @ops(binary_ufuncs)
+    def test_reference_numerics(self, device, dtype, op):
+        if dtype == torch.bfloat16:
+            np_dtype = np.float16
+
+            def to_numpy(t):
+                return np.array(t.tolist(), dtype=np_dtype)
```

Comment:
torch tensors can be converted to NumPy arrays using .numpy():

```
a = t.cpu().numpy().astype(np_dtype)
```

Reply:
This will fail for `bfloat16` since `numpy` does not know this dtype.

Reply:
Good point; I suppose it will have to map the NumPy dtype to a torch dtype and perform the conversion before converting to NumPy